src/runtime.js

```diff
@@ -173,7 +148,18 @@ function adoptShared(global, callback) {
   global.AMP.toggleExperiment = toggleExperiment.bind(null, global);
 
   // Run specific setup for a single-doc or shadow-doc mode.
-  callback(global);
+  callback(global, modules);
+
+  /**
+   * @param {function(!Object)|{n:string, f:function(!Object)}} fnOrStruct
+   */
+  function installModule(fnOrStruct) {
+    if (typeof fnOrStruct == 'function') {
+      fnOrStruct(global.AMP);
```

Comment:
So, it's backward compatible for extensions. Unfortunately, this is NOT backward compatible for runtime. E.g. if an older runtime loads a newer extension (runtime doesn't understand struct, but new extension is using it), we have an issue. Any advice? We could consider deploying just `fnOrStruct` support first and modify extension wrapper a week later?